From 75b840c0859f6c9c119a6f8a1c17b15ef3c7cb58 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Wed, 26 Jul 2017 21:20:00 -0500
Subject: [PATCH 2224/5242] MLK-16089: mtd: gpmi-nand: calculate the correct
 free oob space for large oob layout setting

commit  0a814ffd6a8db0fe24261f1f86dbdba48c27526e from
https://source.codeaurora.org/external/imx/linux-imx.git

for the large oob layout setting, need to calculate the correct free oob
space.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
index 6c392af..57a7081 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
@@ -441,7 +441,8 @@ static int set_geometry_for_large_oob(struct gpmi_nand_data *this)
 
 geo_setting:
 
-	geo->page_size = mtd->writesize + mtd->oobsize;
+	geo->page_size = mtd->writesize + geo->metadata_size +
+		(geo->gf_len * geo->ecc_strength * geo->ecc_chunk_count) / 8;
 	geo->payload_size = mtd->writesize;
 
 	/*
-- 
1.7.9.5

